empirical_photometry.ipynb

```diff
@@ -0,0 +1,448 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import tynt\n",
+    "from synphot.spectrum import SpectralElement\n",
+    "\n",
+    "sdss_bands = ['u', 'g', 'r', 'i', 'z']\n",
+    "\n",
+    "f = tynt.FilterGenerator()\n",
+    "\n",
+    "bandpasses = {}\n",
+    "for band in sdss_bands:\n",
+    "    filt = f.download_true_transmittance('SLOAN/SDSS.' + band)\n",
+    "    bp = SpectralElement(Empirical1D,\n",
+    "                         points=filt.wavelength,\n",
+    "                         lookup_table=filt.transmittance)\n",
+    "    # for boxy bandpasses, tapering them with taper() is recommended\n",
+    "    bandpasses[band] = bp.taper()\n",
+    "\n",
+    "    plt.plot(bp.waveset, bp(bp.waveset), label=band)\n",
+    "plt.legend(loc='upper right')\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also construct a bandpass from a file using <tt>synphot</tt>'s <a href=\"https://synphot.readthedocs.io/en/latest/api/synphot.spectrum.SpectralElement.html#synphot.spectrum.SpectralElement.from_file\">SpectralElement.from_file()</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once we have all of the effects on the source spectrum set, we can combine them together by simply multiplying the SpectralElement objects:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model the observation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from synphot.observation import Observation\n",
+    "\n",
+    "# zero point flux of sdss system\n",
+    "zero_point_star_equiv = u.zero_point_flux(3631.1 * u.Jy)\n",
+    "\n",
+    "fluxs = {}\n",
+    "for band in sdss_bands:\n",
+    "    observation = Observation(spectrum, bandpasses[band], force='taper')\n",
+    "    flux = observation.effstim('Jy')\n",
+    "    fluxs[band] = flux.to(u.nanomaggy, zero_point_star_equiv)\n",
+    "\n",
+    "# # SDSS mags aren't exactly equal to AB mags. Apply corrections\n",
+    "# mags['u'] += 0.04\n",
+    "# mags['z'] -= 0.02"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "u = 8.98588 nmgy\n",
+      "g = 144.40355 nmgy\n",
+      "r = 256.01483 nmgy\n",
+      "i = 397.2675 nmgy\n",
+      "z = 362.74829 nmgy\n"
+     ]
+    }
+   ],
+   "source": [
+    "for band in sdss_bands:\n",
+    "    print(band + ' =', np.round(fluxs[band], 5))"
```

Comment:
Just curious – is there a rationale for 5 decimal places?

Reply:
Ah, no, just a remnant from using magnitudes. I'll fix!